src/control/destination/background/mod.rs

```diff
@@ -288,16 +287,17 @@ where
     /// Tries to reconnect next watch stream. Returns true if reconnection started.
     fn poll_reconnect(&mut self, client: &mut T) -> bool {
         debug_assert!(self.rpc_ready);
-
-        while let Some(auth) = self.dsts.reconnects.pop_front() {
-            if let Some(set) = self.dsts.destinations.get_mut(&auth) {
-                set.query = self.new_query
-                    .query_destination_service_if_relevant(
-                        Some(client),
-                        &auth,
-                        "reconnect",
-                    );
-                return !set.needs_query_capacity();
+        let dsts = &mut self.dsts;
+        while let Some(auth) = dsts.reconnects.pop_front() {
+            if let Some(set) = auth
+                // If the `DestinationSet` corresponding to this authority
+                // has already been dropped, the `Weak` reference to it in
+                // `reconnects` no longer points to a present value. That's
+                // fine, as we no longer need to reconnect that authority.
```

Comment:
nit: prefer to avoid having comments within a statement -- it's jarring.  Also, I think this comment could be pithier:

```
// Reconnect iff the referenced authority still exists in `Destinations`
```